invisible_cities/cities/penthesilea_test.py

```diff
@@ -221,28 +219,41 @@ def test_penthesilea_read_multiple_files(ICDATADIR, output_tmpdir):
 
     penthesilea(**conf)
 
-    with tb.open_file(file_out) as h5out:
-        last_particle_list = h5out.root.MC.extents[:]['last_particle']
-        last_hit_list      = h5out.root.MC.extents[:]['last_hit'     ]
-
-        assert all(x<y for x, y in zip(last_particle_list, last_particle_list[1:]))
-        assert all(x<y for x, y in zip(last_hit_list     , last_hit_list     [1:]))
-
-        with tb.open_file(second_file) as h5second:
-            first_event_out = 4
-            nparticles_in_first_event_out = h5second.root.MC.extents[first_event_out]['last_particle'] - h5second.root.MC.extents[first_event_out - 1]['last_particle']
-            nhits_in_first_event_out      = h5second.root.MC.extents[first_event_out]['last_hit']      - h5second.root.MC.extents[first_event_out - 1]['last_hit'     ]
-
-            nevents_out_in_first_file = 5
-
-            assert last_particle_list[nevents_out_in_first_file] - last_particle_list[nevents_out_in_first_file - 1] == nparticles_in_first_event_out
-            assert last_hit_list     [nevents_out_in_first_file] - last_hit_list     [nevents_out_in_first_file - 1] == nhits_in_first_event_out
+    first_file  = os.path.join(ICDATADIR    ,
+                               "Tl_v1_00_05_nexus_v5_02_08_7bar_pmaps_5evts_0.h5")
+    second_file = os.path.join(ICDATADIR    ,
+                               "Tl_v1_00_05_nexus_v5_02_08_7bar_pmaps_5evts_1.h5")
+
+    particles_in1 = load_mcparticles_df( first_file)
+    hits_in1      = load_mchits_df     ( first_file)
+    particles_in2 = load_mcparticles_df(second_file)
+    hits_in2      = load_mchits_df     (second_file)
+    particles_out = load_mcparticles_df(   file_out)
+    hits_out      = load_mchits_df     (   file_out)
+
+    # back to front beacause that's how the events are.
+    evt_in  = np.concatenate([hits_in2.index.levels[0],
+                              hits_in1.index.levels[0]])
+    first_event_out = 4
+    evt_out = hits_out.index.levels[0]
+    print(evt_in, evt_out)
+    assert all(evt_in[first_event_out:] == evt_out)
+
+    all_hit_in      = pd.concat([hits_in1     ,      hits_in2])
+    pd.testing.assert_frame_equal(all_hit_in.loc[evt_in[first_event_out:]],
+                                  hits_out                                )
+    all_particle_in = pd.concat([particles_in1, particles_in2])
+    pd.testing.assert_frame_equal(all_particle_in.loc[evt_in[first_event_out:]],
+                                  particles_out                                )
```

Comment:
This test is very weird anyway, as I understand we want to check that all the events passed are in output file? @gonzaponte you were the original author of this test?

If I am correct isnt it better to do something like this:
```
input_files = glob(file_in)
events_passed = [14004, 14005, 14006, 14007, 14008, 14009]
all_inp_hits      = pd.concat([load_mchits_df(file_) for file_ in input_files], axis=0)
all_inp_particles = pd.concat([load_mcparticles_df(file_) for file_ in input_files], axis=0)
all_out_hits      = load_mchits_df(file_out)
all_out_particles = load_mcparticles_df(file_out)
assert_frame_equal(all_inp_hits.loc[events_passed, :], all_out_hits)
assert_frame_equal(all_inp_particles.loc[events_passed, :], all_out_particles)
```

ie, instead of hard coding the number of first events from first file that didnt pass filter just hardcode the event numbers that we expect in the output? and later check that the mcinfo is the same?